src/Renderer/ThreeExtended/B3dmLoader.js

```diff
@@ -0,0 +1,70 @@
+import * as THREE from 'three';
+import 'three/examples/js/loaders/GLTFLoader';
+
+function B3dmLoader() {
+    this.glTFLoader = new THREE.GLTFLoader();
+}
+
+const textDecoder = new TextDecoder('utf-8');
+B3dmLoader.prototype.parse = function parse(buffer) {
+    if (!buffer) {
+        throw new Error('No array buffer provided.');
+    }
+
+    const array = new Uint8Array(buffer);
+    const view = new DataView(buffer);
```

Comment:
Using :
```js
// 4 first bytes defined magic type
const array = new Uint8Array(array, 0, 4);
const view = new DataView(buffer, 4);
```
You can remove `byteOffset += 4;` and remove the `subarray` thing.
Actually you could avoid the `array` variable completelty and do:
```js
b3dmHeader.magic = textDecoder.decode(new Uint8Array(buffer, 0, 4));
```
?